charm-seeder/cli.ts

```diff
@@ -84,52 +84,101 @@ async function login() {
   console.log("Logging in");
 
   // Click the first button, "register"
-  await waitForSelectorClick(page, "button[aria-label='register']");
+  await waitForSelectorClick(targetPage, "button[aria-label='register']");
 
   // Click the first button, "register with passphrase"
   await waitForSelectorClick(
-    page,
-    "button[aria-label='method-passphrase']",
+    targetPage,
+    "button[aria-label='register-with-passphrase']",
   );
 
   // Get the mnemonic from textarea.
-  let input = await page.waitForSelector("textarea[aria-label='mnemonic']");
+  let input = await targetPage.waitForSelector("textarea[aria-label='mnemonic']");
   const mnemonic = await input!.evaluate((textarea: HTMLInputElement) =>
     textarea.value
   );
 
   // Click the SECOND button, "continue to login"
-  await waitForSelectorClick(page, "button[aria-label='continue-login']");
+  await waitForSelectorClick(targetPage, "button[aria-label='continue-login']");
 
   // Paste the mnemonic in the input.
-  input = await page.waitForSelector("input[aria-label='enter-passphrase']");
+  input = await targetPage.waitForSelector("input[aria-label='enter-passphrase']");
   await input!.evaluate(
     (input: HTMLInputElement, mnemonic: string) => input.value = mnemonic,
     { args: [mnemonic] },
   );
 
   // Click the only button, "login"
-  await waitForSelectorClick(page, "button[aria-label='login']");
+  await waitForSelectorClick(targetPage, "button[aria-label='login']");
+
+  await targetPage.waitForSelector("#user-avatar");
+}
 
-  await page.waitForSelector("#user-avatar");
+async function login() {
+  await loginWithPage(page);
 }
 
 async function processPrompts() {
   let promptCount = 0;
   console.log(`Processing prompts...`);
 
-  for (const scenario of scenarios) {
-    await page.goto(toolshedUrl);
-    await sleep(1000);
-    let lastCharmId: string | undefined = undefined;
-    for (const step of scenario.steps) {
-      promptCount++;
-      const newCharmId = await processCommand(step, lastCharmId);
-      if (newCharmId) {
-        lastCharmId = newCharmId;
+  // Create a function that processes a single scenario
+  async function processScenario(scenario: Scenario, index: number): Promise<void> {
+    console.log(`[Scenario ${index}] Starting: ${scenario.name}`);
+    
+    // Create a new browser instance for each scenario
+    const scenarioBrowser = await launch({
+      args: ["--window-size=1280,1024"],
+      headless: HEADLESS,
+    });
+    
+    const scenarioPage = await scenarioBrowser.newPage();
```

Comment:
I'm a bit concerned about using multiple tabs for this step - as folks like @ellyxir has already seen timeouts for login / browser flows - due to slowness of access / machine

concurrent generations (happening in deno) seems great - my concern is only browser chonkyness (techinical term)

Reply:
The less a browser is used here, the better; maybe can get away with only using a browser for the final rendering/screenshots